indexing-service/src/main/java/org/apache/druid/indexing/common/actions/CheckPointDataSourceMetadataAction.java

```diff
@@ -68,13 +69,32 @@ public Integer getTaskGroupId()
   }
 
   // For backwards compatibility
-  @Deprecated
   @JsonProperty
-  public SeekableStreamDataSourceMetadata getPreviousCheckPoint()
+  private SeekableStreamDataSourceMetadata getPreviousCheckPoint()
   {
     return checkpointMetadata;
   }
 
+  // For backwards compatibility
+  @JsonProperty
+  private SeekableStreamDataSourceMetadata getCurrentCheckPoint()
+  {
+    return checkpointMetadata;
+  }
+
+  /**
+   * Returning a dummy value so the objects from older versions still work
+   * with current version
+   * TODO : this should be removed in the next release
```

Comment:
I don't think "next release" is right, maybe a few releases down the road. How about: it should be removed when we don't need rolling-update compatibility with version X anymore (please fill in X with the right value).

Reply:
sounds good, i guess X should be 0.15 or earlier